pkg/component/etcd/bootstrap/etcd.go

```diff
@@ -0,0 +1,269 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package bootstrap
+
+import (
+	"context"
+	"net"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/component/etcd/etcd"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+const (
+	metricsPort = 2381
+
+	volumeNameData          = "main-etcd"
+	volumeNameETCDCA        = "etcd-ca"
+	volumeNameServerTLS     = "etcd-server-tls"
+	volumeNameClientTLS     = "etcd-client-tls"
+	volumeNamePeerCA        = "etcd-peer-ca"
+	volumeNamePeerServerTLS = "etcd-peer-server-tls"
+
+	volumeMountPathData          = "/var/etcd/data"
+	volumeMountPathETCDCA        = "/var/etcd/ssl/ca"
+	volumeMountPathServerTLS     = "/var/etcd/ssl/server"
+	volumeMountPathPeerCA        = "/var/etcd/ssl/peer/ca"
+	volumeMountPathPeerServerTLS = "/var/etcd/ssl/peer/server"
+)
+
+// Values is a set of configuration values for the Etcd component.
+type Values struct {
+	// Image is the container image used for Etcd.
+	Image string
+}
+
+// New creates a new instance of DeployWaiter for the Etcd.
+func New(
+	client client.Client,
+	namespace string,
+	secretsManager secretsmanager.Interface,
+	values Values,
+) component.DeployWaiter {
```

Comment:
Change to `Deployer` only?

Reply:
Good point. I was looking into too many `DeployWaiter` and totally forgot `Deployer`.